File: lab-2.3-ByLiza/C#/MyString.cs
using System;
using System.Globalization;

public class MyString : IKeyGenerator
{
    private string content;

    public string Content
    {
        get { return content; }
    }

    public MyString(string input)
    {
        content = input;
    }

    public void CapitalizeWords()
    {
        TextInfo textInfo = CultureInfo.CurrentCulture.TextInfo;
        content = textInfo.ToTitleCase(content.ToLower());

    }

    public string GenerateKey()
    {
        string[] words = content.Split(new string[] { " " }, StringSplitOptions.RemoveEmptyEntries);
        string key = "";

        for (int i = 0; i < words.Length; i++)
        {
            key += words[i][0];
        }

        return key;
    }

}

File: lab-2.3-ByLiza/C++/MyString.cpp
#include "MyString.h"
#include <sstream>
#include <cctype>

MyString::MyString(const std::string& input) : content(input) {}

std::string MyString::GetContent() const {
    return content;
}

std::string MyString::GenerateKey() const {
    std::istringstream iss(content);
    std::string word, key;
    while (iss >> word) {
        if (!word.empty())
            key += word[0];
    }
    return key;
}

void MyString::CapitalizeWords() {
    std::istringstream iss(content);
    std::ostringstream oss;
    std::string word;
    bool first = true;
    while (iss >> word) {
        if (!word.empty())
            word[0] = std::toupper(word[0]);
        if (!first) oss << " ";
        oss << word;
        first = false;
    }
    content = oss.str();
}